test/preview/document_stores/test_memory.py

```diff
@@ -12,3 +15,123 @@ class TestMemoryDocumentStore(DocumentStoreBaseTests):
     @pytest.fixture
     def docstore(self) -> MemoryDocumentStore:
         return MemoryDocumentStore()
+
+    @pytest.mark.unit
+    def test_bm25_retrieval(self, docstore):
+        # Tests if the bm25_retrieval method returns the correct document based on the input query.
+        docs = [
+            Document.from_dict({"content": "Hello world"}),
+            Document.from_dict({"content": "Haystack supports multiple languages"}),
+        ]
+        docstore.write_documents(docs)
+        results = docstore.bm25_retrieval(query="What languages?", top_k=1)
+        assert len(results) == 1
+        assert results[0].content == "Haystack supports multiple languages"
+
+    @pytest.mark.unit
+    def test_bm25_retrieval_with_empty_document_store(self, docstore):
```

Comment:
Just an idea, let's test that it logs about this too? See line 228 of `MemoryDocumentStore`